### PR TITLE
Rubbershot is now very bouncy

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Shotguns have been butt for a while now and don't really have a niche that justifies their clunkiness compared to energy guns and disablers (can't affix tactical lights, have to carry extra ammo for it, can't pass through windows+grilles, must be two handed etc etc), so this PR aims to carve out a niche for them in a fun way that plays into the environments they're used in.

Rubbershot is now much bouncier, has a slightly tighter choke, and has less stamina damage drop off over range, though they also fly 50% slower than before. This way, shotguns will gain a niche in close quarters maint tunnels, where the walls act like a funnel to extend their range, while rifles and beam guns are still much more practical for open areas and general use. It also increases the funny side effect of being able to hit totally unintended bystanders/friendlies or even yourself if you're not careful (though it's still rubbershot so not like it'll do much damage).

[Here's a sample](https://streamable.com/o2wg8c)

## Why It's Good For The Game
Gives a reason to use shotguns that encourages people to take their environment into consideration in deciding whether to break one out. 

## Changelog

:cl: Ryll/Shaps
balance: Rubbershot is now much more bouncy than before, loses damage over range a bit slower, and has a somewhat tighter spread. Its pellets now also fly 50% slower.
fix: Bullets that deal both brute and stamina damage like .38 Rubber will now properly have both their brute and stamina damage reduced after each ricochet, instead of just the brute.
/:cl:

### CODE REVIEW

**Comment by Moltovz** [comment]
Did you consider making pellets that _do_ manage to bounce do less stamina damage? That two-shot from five tiles away because of the bounces is a bit cooky.

**Comment by Ryll-Ryll** [comment]
That's already a feature, the bullet's damage is multiplied by ricochet_decay_damage after each bounce, which is normally 0.7x, but is 0.8x for rubbershot.

.....that being said i just realized that only applies to the main damage var and not the dumb stamina one, so I'm gonna fix that now

**Comment by DrDiasyl** [comment]
Liking this idea. Because shotguns are actually niche after all lethal rounds was removed from armory (For good of course, always hated to see that shotguns get looted the first)

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -367,6 +367,7 @@
 			decayedRange = max(0, decayedRange - reflect_range_decrease)
 			ricochet_chance *= ricochet_decay_chance
 			damage *= ricochet_decay_damage
+			stamina *= ricochet_decay_damage
```
Comment (Iamgoofball) [inline code/modules/projectiles/projectile.dm:370]: detective bouncy rubber rounds nerf 😔

**Comment by TheBonded** [comment]
how many times do they bounce? would it be possible to add a techshell that bounces infinitely with no brute damage?

I just want to make a room in maints that has a bunch of rubber pellets bouncing around indefinitely until someone opens the door and gets hit by 300 shots immediately stamcritting them and breaking all their bones

**Comment by optimumtact** [comment]
>how many times do they bounce? would it be possible to add a techshell that bounces infinitely with no brute damage?

what the hell is wrong with you man

**Comment by TheBonded** [comment]
> > how many times do they bounce? would it be possible to add a techshell that bounces infinitely with no brute damage?
> 
> what the hell is wrong with you man

For starters, I play Space Station 13

**Pull request merged**